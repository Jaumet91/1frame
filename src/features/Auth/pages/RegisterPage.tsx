import { Link } from 'react-router-dom';

import { logoWhite } from '../../../assets/img';
import { startGoogleSignIn } from '../../../store/auth';
import { useAppDispatch } from '../../../store';

export const RegisterPage = () => {
  const dispatch = useAppDispatch();

  const onGoogleSignIn = () => {
    dispatch(startGoogleSignIn());
  };

  return (
    <div className="login">
      <div className="login__containter">
        <div className="login__wrap">
          <Link className="login__logo" to="/news-feed">
            <img
              className="login__pic"
              src={logoWhite}
              alt="login pic"
              aria-label="logo-img"
            />
          </Link>
          <div className="login__info">
            All your favourite games in one place
          </div>
          <div className="games">
            <a className="games__item" href="#">
              <div className="games__preview">
                <img className="games__pic" src="gamesrc" alt="games pic" />
              </div>
              <div className="games_details">
                <div className="games__title">Persona 5</div>
                <div className="games__category">RPG</div>
              </div>
            </a>

            <a className="games__item" href="#">
              <div className="games__preview">
                <img className="games__pic" src="gamesrc" alt="games pic" />
              </div>
              <div className="games_details">
                <div className="games__title">Persona 5</div>
                <div className="games__category">RPG</div>
              </div>
            </a>

            <a className="games__item" href="#">
              <div className="games__preview">
                <img className="games__pic" src="gamesrc" alt="games pic" />
              </div>
              <div className="games_details">
                <div className="games__title">Persona 5</div>
                <div className="games__category">RPG</div>
              </div>
            </a>
          </div>
          <Link className="login__add" to="/search">
            <svg className="icon icon-add-square">
              <use xlinkHref="img/sprite.svg#icon-add-square"></use>
            </svg>
            Discover more
          </Link>
        </div>

        <div className="login__form">
          <Link className="login__close" to="/news-feed">
            <svg className="icon icon-arrow-left">
              <use xlinkHref="img/sprite.svg#icon-arrow-left"></use>
            </svg>
          </Link>
          <h3 className="login__title h3">Sign up</h3>
          <div className="login__line">
            <div className="login__text">Already a user</div>
            <Link className="login__link" to="/auth/login">
              Login now
            </Link>
          </div>

          <div className="login__row">
            <div className="login__col">
              <div className="field">
                <label htmlFor="FirstName" className="field__label">
                  First name
                </label>
                <div className="field__wrap">
                  <input
                    id="FirstName"
                    name="FirstName"
                    className="field__input"
                    type="text"
                  />
                </div>
              </div>
            </div>
            <div className="login__col">
              <div className="field">
                <label htmlFor="LastName" className="field__label">
                  Last name
                </label>
                <div className="field__wrap">
                  <input
                    id="LastName"
                    name="LastName"
                    className="field__input"
                    type="text"
                  />
                </div>
              </div>
            </div>
          </div>

          <div className="field">
            <label htmlFor="EmailAddress" className="field__label">
              Email Address
            </label>
            <div className="field__wrap">
              <input
                id="EmailAddress"
                name="EmailAddress"
                className="field__input"
                type="email"
              />
            </div>
          </div>

          <div className="field">
            <label htmlFor="Password" className="field__label">
              Password
            </label>
            <div className="field__wrap">
              <input
                id="Password"
                name="Password"
                className="field__input"
                type="password"
              />
            </div>
          </div>

          <button className="login__btn btn btn_purple btn_wide">
            Continue
          </button>
          <div className="login__or">Sign up by Google</div>
          <button
            className="login__btn btn btn_blue btn_wide"
            onClick={onGoogleSignIn}
          >
            Google Account
          </button>
        </div>
      </div>
    </div>
  );
};
