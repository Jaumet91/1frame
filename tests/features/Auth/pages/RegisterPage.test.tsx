import React from 'react';
import { render, screen } from '@testing-library/react';
import { RegisterPage } from '../../../../src/features/Auth/pages/RegisterPage';
import { MemoryRouter } from 'react-router-dom';

describe('Tests in <RegisterPage />', () => {
  test('should render the register page by default', () => {
    const { container } = render(
      <MemoryRouter>
        <RegisterPage />
      </MemoryRouter>
    );

    const logoImg: HTMLImageElement = screen.getByRole('img', {
      name: 'logo-img'
    });

    expect(screen.getAllByText('All your favourite games in one place'));
    expect(container).toMatchSnapshot();
  });
});
