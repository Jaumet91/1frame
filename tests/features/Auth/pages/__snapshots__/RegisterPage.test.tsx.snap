// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests in <RegisterPage /> should render the register page by default 1`] = `
<div>
  <div
    class="login"
  >
    <div
      class="login__containter"
    >
      <div
        class="login__wrap"
      >
        <a
          class="login__logo"
          href="/news-feed"
        >
          <img
            alt="login pic"
            aria-label="logo-img"
            class="login__pic"
            src=""
          />
        </a>
        <div
          class="login__info"
        >
          All your favourite games in one place
        </div>
        <div
          class="games"
        >
          <a
            class="games__item"
            href="#"
          >
            <div
              class="games__preview"
            >
              <img
                alt="games pic"
                class="games__pic"
                src="gamesrc"
              />
            </div>
            <div
              class="games_details"
            >
              <div
                class="games__title"
              >
                Persona 5
              </div>
              <div
                class="games__category"
              >
                RPG
              </div>
            </div>
          </a>
          <a
            class="games__item"
            href="#"
          >
            <div
              class="games__preview"
            >
              <img
                alt="games pic"
                class="games__pic"
                src="gamesrc"
              />
            </div>
            <div
              class="games_details"
            >
              <div
                class="games__title"
              >
                Persona 5
              </div>
              <div
                class="games__category"
              >
                RPG
              </div>
            </div>
          </a>
          <a
            class="games__item"
            href="#"
          >
            <div
              class="games__preview"
            >
              <img
                alt="games pic"
                class="games__pic"
                src="gamesrc"
              />
            </div>
            <div
              class="games_details"
            >
              <div
                class="games__title"
              >
                Persona 5
              </div>
              <div
                class="games__category"
              >
                RPG
              </div>
            </div>
          </a>
        </div>
        <a
          class="login__add"
          href="/search"
        >
          <svg
            class="icon icon-add-square"
          >
            <use
              xlink:href="img/sprite.svg#icon-add-square"
            />
          </svg>
          Discover more
        </a>
      </div>
      <div
        class="login__form"
      >
        <a
          class="login__close"
          href="/news-feed"
        >
          <svg
            class="icon icon-arrow-left"
          >
            <use
              xlink:href="img/sprite.svg#icon-arrow-left"
            />
          </svg>
        </a>
        <h3
          class="login__title h3"
        >
          Sign up
        </h3>
        <div
          class="login__line"
        >
          <div
            class="login__text"
          >
            Already a user
          </div>
          <a
            class="login__link"
            href="/auth/login"
          >
            Login now
          </a>
        </div>
        <div
          class="login__row"
        >
          <div
            class="login__col"
          >
            <div
              class="field"
            >
              <label
                class="field__label"
                for="FirstName"
              >
                First name
              </label>
              <div
                class="field__wrap"
              >
                <input
                  class="field__input"
                  id="FirstName"
                  name="FirstName"
                  type="text"
                />
              </div>
            </div>
          </div>
          <div
            class="login__col"
          >
            <div
              class="field"
            >
              <label
                class="field__label"
                for="LastName"
              >
                Last name
              </label>
              <div
                class="field__wrap"
              >
                <input
                  class="field__input"
                  id="LastName"
                  name="LastName"
                  type="text"
                />
              </div>
            </div>
          </div>
        </div>
        <div
          class="field"
        >
          <label
            class="field__label"
            for="EmailAddress"
          >
            Email Address
          </label>
          <div
            class="field__wrap"
          >
            <input
              class="field__input"
              id="EmailAddress"
              name="EmailAddress"
              type="email"
            />
          </div>
        </div>
        <div
          class="field"
        >
          <label
            class="field__label"
            for="Password"
          >
            Password
          </label>
          <div
            class="field__wrap"
          >
            <input
              class="field__input"
              id="Password"
              name="Password"
              type="password"
            />
          </div>
        </div>
        <button
          class="login__btn btn btn_purple btn_wide"
        >
          Continue
        </button>
        <div
          class="login__or"
        >
          Sign up by Google
        </div>
        <button
          class="login__btn btn btn_blue btn_wide"
        >
          Google Account
        </button>
      </div>
    </div>
  </div>
</div>
`;
